torch/csrc/autograd/engine.cpp

```diff
@@ -691,17 +691,29 @@ void validate_outputs(
       // AT_ERROR(format_error(ss.str()));
       continue;
     }
-    if (!grad.sizes().equals(metadata.shape())) {
```

Comment:
spaghetti but need some torch.Size vs NestedSize(torch.Tensor) agnostic way of checking equality